### ISSUE TITLE
yaml文件参数问题

### ISSUE THREAD

**Issue by wstchhwp** [PaddlePaddle/PaddleSeg#497]
你好，我想问下如何查询yaml文件里面有哪些参数？参数都可以取哪些值？比如slover这块配置都有哪些参数，lr学习率有哪些方式？
有没有命令可以查询。谢谢回答

**Comment by nepeplwu** [comment]
@wstchhwp 可以参考这份文档：https://github.com/PaddlePaddle/PaddleSeg/blob/release/v0.6.0/docs/config.md

**Comment by wstchhwp** [comment]
这份文档我看过，自己还有些迷茫，因为就比如说学习率只给了三种，paddleseg只封装了三种吗？优化算法是Adam,sgd两种，只封装了两种嘛？
一下是运行时产生的，有些配置你给的链接里没有
{'AUG': {'AUG_METHOD': 'unpadding',
         'FIX_RESIZE_SIZE': (256, 256),
         'FLIP': False,
         'FLIP_RATIO': 0.5,
         'INF_RESIZE_VALUE': 500,
         'MAX_RESIZE_VALUE': 600,
         'MAX_SCALE_FACTOR': 2.0,
         'MIN_RESIZE_VALUE': 400,
         'MIN_SCALE_FACTOR': 0.5,
         'MIRROR': True,
         'RICH_CROP': {'ASPECT_RATIO': 0.33,
                       'BLUR': False,
                       'BLUR_RATIO': 0.1,
                       'BRIGHTNESS_JITTER_RATIO': 0.5,
                       'CONTRAST_JITTER_RATIO': 0.5,
                       'ENABLE': False,
                       'MAX_ROTATION': 15,
                       'MIN_AREA_RATIO': 0.5,
                       'SATURATION_JITTER_RATIO': 0.5},
         'SCALE_STEP_SIZE': 0.25,
         'TO_RGB': False},
 'BATCH_SIZE': 32,
 'DATALOADER': {'BUF_SIZE': 256, 'NUM_WORKERS': 8},
 'DATASET': {'DATA_DIM': 3,
             'DATA_DIR': 'PaddleSeg/dataset/rs_data',
             'IGNORE_INDEX': 255,
             'IMAGE_TYPE': 'rgb',
             'NUM_CLASSES': 7,
             'PADDING_VALUE': [127.5, 127.5, 127.5],
             'SEPARATOR': ' ',
             'TEST_FILE_LIST': 'PaddleSeg/dataset/rs_data/testA_list.txt',
             'TEST_TOTAL_IMAGES': 10001,
             'TRAIN_FILE_LIST': 'PaddleSeg/dataset/rs_data/train_list.txt',
             'TRAIN_TOTAL_IMAGES': 143062,
             'VAL_FILE_LIST': 'PaddleSeg/dataset/rs_data/val_list.txt',
             'VAL_TOTAL_IMAGES': 2919,
             'VIS_FILE_LIST': 'PaddleSeg/dataset/rs_data/testA_list.txt'},
 'EVAL_CROP_SIZE': (256, 256),
 'FREEZE': {'MODEL_FILENAME': '__model__',
            'PARAMS_FILENAME': '__params__',
            'SAVE_DIR': 'freeze_model'},
 'MEAN': [0.5, 0.5, 0.5],
 'MODEL': {'BN_MOMENTUM': 0.99,
           'DEEPLAB': {'ASPP_WITH_SEP_CONV': True,
                       'BACKBONE': 'xception_65',
                       'BACKBONE_LR_MULT_LIST': None,
                       'DECODER': {'CONV_FILTERS': 256,
                                   'OUTPUT_IS_LOGITS': False,
                                   'USE_SUM_MERGE': False},
                       'DECODER_USE_SEP_CONV': True,
                       'DEPTH_MULTIPLIER': 1.0,
                       'ENABLE_DECODER': True,
                       'ENCODER': {'ADD_IMAGE_LEVEL_FEATURE': True,
                                   'ASPP_CONVS_FILTERS': 256,
                                   'ASPP_RATIOS': None,
                                   'ASPP_WITH_CONCAT_PROJECTION': True,
                                   'ASPP_WITH_SE': False,
                                   'POOLING_CROP_SIZE': None,
                                   'POOLING_STRIDE': [1, 1],
                                   'SE_USE_QSIGMOID': False},
                       'ENCODER_WITH_ASPP': True,
                       'OUTPUT_STRIDE': 16},
           'DEFAULT_EPSILON': 1e-05,
           'DEFAULT_GROUP_NUMBER': 32,
           'DEFAULT_NORM_TYPE': 'bn',
           'FP16': False,
           'HRNET': {'STAGE2': {'NUM_CHANNELS': [40, 80], 'NUM_MODULES': 1},
                     'STAGE3': {'NUM_CHANNELS': [40, 80, 160],
                                'NUM_MODULES': 4},
                     'STAGE4': {'NUM_CHANNELS': [40, 80, 160, 320],
                                'NUM_MODULES': 3}},
           'ICNET': {'DEPTH_MULTIPLIER': 0.5, 'LAYERS': 50},
           'MODEL_NAME': 'pspnet',
           'MULTI_LOSS_WEIGHT': [1.0],
           'OCR': {'OCR_KEY_CHANNELS': 256, 'OCR_MID_CHANNELS': 512},
           'PSPNET': {'DEPTH_MULTIPLIER': 1, 'LAYERS': 101},
           'SCALE_LOSS': 'DYNAMIC',
           'UNET': {'UPSAMPLE_MODE': 'bilinear'}},
 'NUM_TRAINERS': 1,
 'SLIM': {'KNOWLEDGE_DISTILL': False,
          'KNOWLEDGE_DISTILL_IS_TEACHER': False,
          'KNOWLEDGE_DISTILL_TEACHER_MODEL_DIR': '',
          'NAS_ADDRESS': '',
          'NAS_IS_SERVER': True,
          'NAS_PORT': 23333,
          'NAS_SEARCH_STEPS': 100,
          'NAS_SPACE_NAME': '',
          'NAS_START_EVAL_EPOCH': 0,
          'PREPROCESS': False,
          'PRUNE_PARAMS': '',
          'PRUNE_RATIOS': []},
 'SOLVER': {'BEGIN_EPOCH': 1,
            'CROSS_ENTROPY_WEIGHT': None,
            'DECAY_EPOCH': [10, 20],
            'GAMMA': 0.1,
            'LOSS': ['softmax_loss'],
            'LOSS_WEIGHT': {'BCE_LOSS': 1,
                            'DICE_LOSS': 1,
                            'LOVASZ_HINGE_LOSS': 1,
                            'LOVASZ_SOFTMAX_LOSS': 1,
                            'SOFTMAX_LOSS': 1},
            'LR': 0.01,
            'LR_POLICY': 'poly',
            'LR_WARMUP': False,
            'LR_WARMUP_STEPS': 2000,
            'MOMENTUM': 0.9,
            'MOMENTUM2': 0.999,
            'NUM_EPOCHS': 50,
            'OPTIMIZER': 'Adam',
            'POWER': 0.9,
            'WEIGHT_DECAY': 4e-05},
 'STD': [0.5, 0.5, 0.5],
 'TEST': {'TEST_MODEL': 'PaddleSeg/saved_model/pspnet_optic/7'},
 'TRAIN': {'MODEL_SAVE_DIR': 'PaddleSeg/saved_model/pspnet_optic/',
           'PRETRAINED_MODEL_DIR': 'PaddleSeg/pretrained_model/pspnet50_bn_cityscapes/',
           'RESUME_MODEL_DIR': 'PaddleSeg/saved_model/pspnet_optic/5',
           'SNAPSHOT_EPOCH': 1,
           'SYNC_BATCH_NORM': False},
 'TRAINER_ID': 0,
 'TRAIN_CROP_SIZE': (256, 256)}

**Comment by wstchhwp** [comment]
@nepeplwu 见楼上

**Comment by nepeplwu** [comment]
@wstchhwp 
* 学习率只给了三种，paddleseg只封装了三种吗
是的

* 优化算法是Adam,sgd两种，只封装了两种嘛
是的

* 一下是运行时产生的，有些配置你给的链接里没有
有些配置确实在文档里面缺失了，我们后续会补充上，目前主要是哪个配置不清楚用途？

**Comment by wstchhwp** [comment]
@nepeplwu 
暂时都看懂配置了，希望更新paadleseg由动态图实现，这样就可以改源码了，谢谢！

**Comment by nepeplwu** [comment]
@wstchhwp 感谢关注，动态图版本的PaddleSeg已经在开发中了，将会在0.7.0版本中作为体验功能开放出来
